standard/standard_9.php

```diff
@@ -445,7 +445,6 @@ function array_intersect_uassoc(
  * array1 that are present in all the arguments.
  * @meta
  */
-#[Pure]
```

Comment:
IMO this change is incorrect as well. The function doesn't depend on global state and doesn't have any side effects. it just computes the intersection of arrays with User-defined comparison for values and User-defined comparison for keys
```
function value_compare($a, $b) {
    return strcmp($a, $b);
}
function key_compare($a, $b) {
    return strcmp($a, $b);
}

$array1 = ["a" => "apple", "b" => "banana"];
$array2 = ["a" => "apple", "b" => "berry"];

$result = array_uintersect_uassoc($array1, $array2, "value_compare", "key_compare");
print_r($result); // Only "a" => "apple" matches both value and key
```
Please correct me if I'm wrong.

Reply:
This is not straight forward because in your example this function is pure, but if you use a comparison function that isn't pure then this function is not pure.

The convention in this package seems to be that any function/method taking a callback is not marked as pure.

Reply:
Right. The function can produce side effects via callback. Thank you!